chat.js

```diff
@@ -948,3 +992,39 @@ Chat.toDurationString = function (number, options) {
 	}
 	return parts.slice(positiveIndex).reverse().map((value, index) => value ? value + " " + unitNames[index] + (value > 1 ? "s" : "") : "").reverse().slice(0, precision).join(" ").trim();
 };
+
+/**
+ * @param  {string} str
+ * @return {string}
+ */
+Chat.md5 = function (str) {
+	 let generator = crypto.createHash('md5');
+	 generator.update(str);
+	 return generator.digest('hex');
+};
+
+/**
+ * @param {Room} room
+ * @param {string} checksum
+ */
+Chat.r9kLoadPermanent = function (room) {
```

Comment:
Never mind, I thought `Chat.r9kAppendPermanent` was writing instead of appending for some stupid reason